ludwig/modules/reduction_modules.py

```diff
@@ -35,6 +35,22 @@ def __init__(self, reduce_mode=None, **kwargs):
     def forward(self, inputs, mask=None):
         return self._reduce_obj(inputs, mask=mask)
 
+    def infer_output_shape(self, input_shape):
```

Comment:
This method seems to be doing a lot of what the `output_shape` property in a `LudwigModule` does.

You could make the parent class of `SequenceReducer` a `LudwigModule`, and add `input_shape` and `output_shape` properties. The `output_shape` would have the same logic that's implemented in `infer_output_shape`, and the `input_shape` can return a dummy tensor.

Reply:
Additionally, can you update parts of the code that use `SequenceReducer` and then manually infer the shape of the output after reduction to use the `output_shape` property of the `SequenceReducer` instead?

E.g. here's some [code](https://github.com/ludwig-ai/ludwig/blob/389c1aa443ddc0b5be92981aa9811421d58491bd/ludwig/encoders/sequence_encoders.py#L793) that can use `SequenceReducer.output_shape`.

Reply:
Haha, I missed LudwigModule._compute_output_shape.   However, the sequence reducer does not know the sequence length ahead of time, so I can’t fully specify output shape having input shape passed in somehow.